app/views/partials/_template_component_question_menu.html.erb

```diff
@@ -0,0 +1,20 @@
+<script type="text/html"
+        data-component-template="ComponentQuestionMenu"
+        data-activator-text="<%= t('question.menu.activator') %>">
+  <ul class="govuk-navigation component-activated-menu">
+    <li data-action="required"><span><%= t('question.menu.required')%></span></li>
+    <li data-action="remove"
+        data-api-path="<%= URI.decode_www_form_component(api_service_page_question_destroy_message_path(service.service_id, @page.uuid, '#{question_uuid}', method: :delete)) -%>">
+      <span class="destructive"><%= t('question.menu.remove')%></span>
+    </li>
+
+    <% @page.components.each do |component| %>
+      <% enabled_validations(component).each do |validation| %>
+        <li data-action="<%= validation %>"
+            data-api-path="<%= URI.decode_www_form_component(api_service_page_component_validations_path(service.service_id, @page.uuid,'#{question_uuid}', validation)) -%>">
+          <span><%= t("question.menu.#{validation}") %></span>
```

Comment:
@chrispymm to be honest the `component` is available here so we could just do

`api_service_page_component_validations_path(service.service_id, @page.uuid, component.uuid, validation)`

straight away instead of having to do the find and replace in JS